### PR TITLE
[csharp][generichost] Remove unnecessary parenthesis that causes build issue

### PR DESCRIPTION
**Pull request by lewinskimaciej**
Removes unnecessary parenthesis that causes a build error:
image

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@devhl-labs

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/csharp/libraries/generichost/api.mustache [hunk 1]
```diff
@@ -397,7 +397,7 @@ namespace {{packageName}}.{{apiPackage}}
                     {{^required}}
 
                     if ({{paramName}}.IsSet)
-                        uriBuilderLocalVar.Path = uriBuilderLocalVar.Path + $"/{ Uri.EscapeDataString({{paramName}}.Value).ToString()) }";
+                        uriBuilderLocalVar.Path = uriBuilderLocalVar.Path + $"/{ Uri.EscapeDataString({{paramName}}.Value).ToString() }";
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/csharp/libraries/generichost/api.mustache:400]: this part or code block is for optional path parameters.

from what I know, path parameters should be required all the time.

can you share the use cases in which path parameter is optional?

cc @devhl-labs

Comment (devhl-labs) [reply modules/openapi-generator/src/main/resources/csharp/libraries/generichost/api.mustache:400]: That appears to be correct. Looks like the {{^required}} section can be removed.

https://swagger.io/docs/specification/describing-parameters/#path-parameters:~:text=Also%20remember%20to%20add%C2%A0required%3A%20true%2C%20because%20path%20parameters%20are%20always%20required.

Comment (lewinskimaciej) [reply modules/openapi-generator/src/main/resources/csharp/libraries/generichost/api.mustache:400]: Yeah this is right, we actually had an path param that was marked as optional(not on purpose, was supposed to be a query param, fixed already).
I didn't really mean to change any logic to the generation, just the parenthesis that was causing a build error there if it fell into that case.

Comment (wing328) [reply modules/openapi-generator/src/main/resources/csharp/libraries/generichost/api.mustache:400]: 👌 

I've filed https://github.com/OpenAPITools/openapi-generator/pull/17525 instead to remove optional path parameters so closing this one.

**Pull request closed without merging**